apps/full-site-editing/full-site-editing-plugin/blocks/content-renderer/index.js

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType } from '@wordpress/blocks';
+import { IconButton, Placeholder, Toolbar } from '@wordpress/components';
+import { withState } from '@wordpress/compose';
+import { BlockControls } from '@wordpress/editor';
+import { Fragment, RawHTML } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import PostAutocomplete from '../../components/post-autocomplete';
+import fetchPost from '../../lib/fetch-post';
+import './style.scss';
+
+const setSelectedPost = async ( attributes, setState ) => {
+	const { selectedPostId, selectedPostType } = attributes;
+	const selectedPost = await fetchPost( selectedPostId, selectedPostType );
+	setState( {
+		selectedPost,
+	} );
+};
+
+const edit = withState( {
```

Comment:
It's conventional in core for a block's `edit` method to live in its own file. We should probably try to follow those when needed for future core contribution possibilities.

Reply:
This is a good point.
I'll also update the Content Preview block accordingly.